app/livechat/server/api/v1/room.js

```diff
@@ -179,3 +182,58 @@ API.v1.addRoute('livechat/room.forward', { authRequired: true }, {
 		API.v1.success(Meteor.runAsUser(this.userId, () => Meteor.call('livechat:transfer', this.bodyParams)));
 	},
 });
+
+API.v1.addRoute('livechat/room.visitor', { authRequired: true }, {
+	put() {
+		try {
+			check(this.bodyParams, {
+				rid: String,
+				oldVisitorId: String,
+				newVisitorId: String,
+			});
+
+			const user = Promise.await(Users.findOneById(this.userId));
+			if (!user) {
+				throw new Error('error-user-not-found');
+			}
+
+			if (!Promise.await(hasPermissionAsync(this.userId, 'change-livechat-room-visitor'))) {
```

Comment:
Let's check this permission inside the `Livechat.changeRoomVisitor` method.

Reply:
I've moved the permission check